sql/procedure/user_behaviour_revenue_insights.sqlx

```diff
@@ -0,0 +1,199 @@
+-- Copyright 2023 Google LLC
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Setting procedure to lookback from the day before `input_date` until the day before `end_date`
+SET input_date = DATE_SUB(input_date, INTERVAL 1 DAY);
+
+
+CREATE OR REPLACE TEMP TABLE prompts_per_day AS
+SELECT DISTINCT
+calculation_base_date as feature_date,
+CONCAT("You're a Google Analytics 4 Marketing Analyst. Your objective is to extract important insights from all the user behaviour and revenue metrics calculated for the users. \n",
+    "Here are the user behaviour and revenue metrics calculated over the past 30 days from a base date.\n\n",
+    "Metrics JSON:\n",
+    TO_JSON_STRING(t),
+    "\n\n",
+    "Here are a the metrics descriptions:\n",
+    (SELECT STRING_AGG(FORMAT("%s (%s): %s", column_name, data_type, description), "\n") FROM `marketing-data-engine-demo`.feature_store.INFORMATION_SCHEMA.COLUMN_FIELD_PATHS WHERE table_name = 'user_scoped_metrics' AND column_name NOT IN ('processed_timestamp', 'feature_date')),
+    "\n\n"
+) as prompt
+FROM (SELECT DISTINCT  feature_date as calculation_base_date, * EXCEPT(processed_timestamp, feature_date) FROM `marketing-data-engine-demo.feature_store.user_scoped_metrics` WHERE feature_date = input_date ORDER BY feature_date DESC) t
+;
```

Comment:
Hi @chmstimoteo and @kingman, thanks for the great repository! I was trying to use it today and got some error messages. I suspect this line and the subsequent entries in this file should perhaps use `{{project_id}}` rather than the hardcoded marketing-data-engine-demo?